tcplistener.go

```diff
@@ -147,6 +146,13 @@ func handleListenedConn(address string, conn *net.TCPConn, headerByteSize int, m
 	// never be any resultant cross-contamination from earlier runs of the loop.
 	headerBuffer := make([]byte, headerByteSize)
 	dataBuffer := make([]byte, maxMessageSize)
+	// Start an asyncrhonous call that will wait on the shutdown channel, and then close
+	// the connection. This will let us respond to the shutdown but also not incur
+	// a cost for checking the channel on each run of the loop
+	go func(c *net.TCPConn, s chan struct{}) {
+		<-s
+		c.Close()
+	}(conn, sdChan)
```

Comment:
You can reference `conn` and `sdChan` inside the `go func() { ... }`, they'll get picked up in a closure.